### DISCUSSION TITLE
Action as folder

### DISCUSSION THREAD

**Discussion by noellealarie** [rvion/CushyStudio#38] [Ideas]
So hear me out.

Instead of an action being a single file, make it a folder so it can have a separate config file, preview images (for example), and where drafts can be saved for easy sharing.

And the Action List, instead of showing the folders inside Actions, each action could have a category parameter in the config.

**Comment by rvion** [comment]
the goal is great, and I agree that actions can't 100% always be single files:
    - actions in a same action pack need to share metadata (like author info)
    - actions may want to include documentation or example illustrations, or prestes

what I had in mind was to let typescript file resolution handle this; but this may be the wrong way to go. I'm really open to suggestions there.

I'm planning to work and release the marketplace this week, so this is a great moment to experiment with various ideas and ensure all needs of actions creators are covered.

I'll use this repository: https://github.com/rvion/cushy-example-actions to showcase various ways to package and modularize actions in a maintainable way
